shoulder/__main__.py

```diff
@@ -27,11 +27,13 @@
 from shoulder.parser import *
 from shoulder.generator import *
 from shoulder.filter import filter_all
-from shoulder.transform import transform_all
+from shoulder.transform import *
 
 regs = parse_registers(config.xml_register_dir)
 logger.debug("Registers parsed: " + str(len(regs)))
 
+regs = transforms["remove_reserved_0"].transform(regs)
```

Comment:
Yeah, this one is a little awkward. There are quite a few registers that have overlapping fields with reserved fields. For example, in `AArch64-spsr_el1.xml` there is a field defined at 24 and also reserved 0 at 24:
```
 <field id="DIT_24_24_1" is_variable_length="False" has_partial_fieldset="False" is_linked_to_partial_fieldset="False" is_access_restriction_possible="False" is_constant_value="False">
            <field_name>DIT</field_name>
         <field_msb>24</field_msb>
         <field_lsb>24</field_lsb>
         <field_description order="before">

   <para>Data Independent Timing. Set to the value of PSTATE.DIT on taking an exception to EL1, and copied to PSTATE.DIT on executing an exception return operation in EL1.</para>

         </field_description>
         <field_values>


         </field_values>
           <field_resets>

     <field_reset>

       <field_reset_standard_text>U</field_reset_standard_text>

     </field_reset>
 </field_resets>
             <fields_condition>When ARMv8.4-DIT is implemented</fields_condition>
       </field>
       <field id="0_24_24_2" is_variable_length="False" has_partial_fieldset="False" is_linked_to_partial_fieldset="False" is_access_restriction_possible="False" is_constant_value="False">
            <field_name>0</field_name>
         <field_msb>24</field_msb>
         <field_lsb>24</field_lsb>
         <field_description order="before">
             <para>Reserved, <arm-defined-word>RES0</arm-defined-word>.</para>
         </field_description>
         <field_values>
         </field_values>
       </field>
       <field id="SSBS_23_23_1" is_variable_length="False" has_partial_fieldset="False" is_linked_to_partial_fieldset="False" is_access_restriction_possible="False" is_constant_value="False">
            <field_name>SSBS</field_name>
         <field_msb>23</field_msb>
         <field_lsb>23</field_lsb>
         <field_description order="before">
```

These are causing these registers to be invalid since there are overlapping fields. We should probably figure a solution to this, but for the time being, I just transform them before they are filtered out.